File: mobile/lib/shared/models/event.dart
class Event {
  final int id;
  final String titleEn;
  final String? titleAr;
  final String slug;
  final String descriptionEn;
  final String? descriptionAr;
  final DateTime startDate;
  final DateTime endDate;
  final String category;
  final String? coverImage;
  final List<String>? tags;
  final String status;
  final bool hasTickets;
  final bool isFree;
  final int viewCount;
  final int attendingCount;
  final int saveCount;
  final String? venueName;
  final String? venueCity;
  final String? organizerName;
  final double? minPrice;

  Event({
    required this.id,
    required this.titleEn,
    this.titleAr,
    required this.slug,
    required this.descriptionEn,
    this.descriptionAr,
    required this.startDate,
    required this.endDate,
    required this.category,
    this.coverImage,
    this.tags,
    required this.status,
    required this.hasTickets,
    required this.isFree,
    required this.viewCount,
    required this.attendingCount,
    required this.saveCount,
    this.venueName,
    this.venueCity,
    this.organizerName,
    this.minPrice,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    final venue = json['venue'] as Map<String, dynamic>?;
    final organizer = json['organizer'] as Map<String, dynamic>?;
    double? minPrice;
    final ticketTypes = json['ticket_types'] as List<dynamic>?;
    if (ticketTypes != null && ticketTypes.isNotEmpty) {
      minPrice = (ticketTypes.map((t) => (t as Map)['price'] as num).reduce(
            (a, b) => a < b ? a : b,
          ) as num)
          .toDouble();
    }

    return Event(
      id: json['id'] as int,
      titleEn: json['title_en'] as String,
      titleAr: json['title_ar'] as String?,
      slug: json['slug'] as String,
      descriptionEn: json['description_en'] as String,
      descriptionAr: json['description_ar'] as String?,
      startDate: DateTime.parse(json['start_date'] as String),
      endDate: DateTime.parse(json['end_date'] as String),
      category: json['category'] as String,
      coverImage: json['cover_image'] as String?,
      tags: (json['tags'] as List<dynamic>?)?.map((t) => t as String).toList(),
      status: json['status'] as String? ?? 'published',
      hasTickets: json['has_tickets'] as bool? ?? false,
      isFree: json['is_free'] as bool? ?? false,
      viewCount: json['view_count'] as int? ?? 0,
      attendingCount: json['attending_count'] as int? ?? 0,
      saveCount: json['save_count'] as int? ?? 0,
      venueName: venue?['name_en'] as String?,
      venueCity: venue?['city'] as String?,
      organizerName: organizer?['name_en'] as String?,
      minPrice: minPrice,
    );
  }

  String get displayTitle => titleAr ?? titleEn;
  String get displayDescription => descriptionAr ?? descriptionEn;
}

File: backend/app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Str;

class Event extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'organizer_id', 'venue_id', 'created_by_user_id',
        'title_en', 'title_ar', 'slug',
        'description_en', 'description_ar',
        'start_date', 'end_date', 'category',
        'cover_image', 'gallery', 'tags',
        'status', 'has_tickets', 'is_free',
        'view_count', 'attending_count', 'save_count',
    ];

    protected function casts(): array
    {
        return [
            'start_date' => 'datetime',
            'end_date' => 'datetime',
            'gallery' => 'json',
            'tags' => 'json',
            'has_tickets' => 'boolean',
            'is_free' => 'boolean',
        ];
    }

    protected static function booted(): void
    {
        static::creating(function (Event $event) {
            if (empty($event->slug)) {
                $base = Str::slug($event->title_en);
                $slug = $base;
                $count = 2;
                while (Event::where('slug', $slug)->exists()) {
                    $slug = $base . '-' . $count++;
                }
                $event->slug = $slug;
            }
        });
    }

    public function organizer(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(Organizer::class);
    }

    public function venue(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(Venue::class);
    }

    public function createdBy(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by_user_id');
    }

    public function ticketTypes(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(TicketType::class);
    }

    public function orders(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(Order::class);
    }

    public function tickets(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(Ticket::class);
    }

    public function comments(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(Comment::class);
    }

    public function userActions(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(UserEventAction::class);
    }

    public function promoCodes(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(PromoCode::class);
    }

    public function scopePublished($query)
    {
        return $query->where('status', 'published');
    }

    public function scopeUpcoming($query)
    {
        return $query->where('start_date', '>=', now());
    }

    public function scopeByCategory($query, string $category)
    {
        return $query->where('category', $category);
    }

    public function scopeSearch($query, string $term)
    {
        return $query->where(function ($q) use ($term) {
            $q->where('title_en', 'like', "%{$term}%")
              ->orWhere('title_ar', 'like', "%{$term}%")
              ->orWhere('description_en', 'like', "%{$term}%")
              ->orWhere('description_ar', 'like', "%{$term}%");
        });
    }

    public function incrementViewCount(): void
    {
        $this->increment('view_count');
    }

    public function cheapestTicket(): ?TicketType
    {
        return $this->ticketTypes()
            ->where('status', 'available')
            ->orderBy('price')
            ->first();
    }

    public function isUpcoming(): bool
    {
        return $this->start_date->isFuture();
    }
}
